api/selftest.py

```diff
@@ -1,35 +1,54 @@
 import sys
+from typing import Iterable, List, Optional, Tuple, Union
 
 from sqlalchemy.orm.session import Session
 
 from core.config import IntegrationException
-from core.model import ExternalIntegration, LicensePool
+from core.model import Collection, ExternalIntegration, Library, LicensePool, Patron
 from core.opds_import import OPDSImporter, OPDSImportMonitor
 from core.scripts import LibraryInputScript
 from core.selftest import HasSelfTests as CoreHasSelfTests
 from core.selftest import SelfTestResult
+from core.util.problem_detail import ProblemDetail
 
 from .authenticator import LibraryAuthenticator
 from .circulation import CirculationAPI
 from .feedbooks import FeedbooksImportMonitor, FeedbooksOPDSImporter
 
 
+class _NoValidLibrarySelfTestPatron(Exception):
```

Comment:
Should this be a subclass of [`BaseError`](https://github.com/ThePalaceProject/circulation-core/blob/3e4f3395409bf18557d7123fb070a321301330c1/exceptions.py#L1)? Might be good to keep all of our exception having a common base class if we can.

If its meant to be used only within `HasSelfTests` should it be defined in that class?